client/app/main/student/student.controller.js

```diff
@@ -109,6 +109,50 @@ function StudentCtrl($scope, $stateParams, Intervention, Modal, Outreach,
       toastr.error(err);
     });
   };
+  $scope.toggleArchiveOutreach = function(outreach) {
+    var oldValue = outreach.archived;
+    var promise = Outreach.toggleArchive({
+      id: outreach._id
+    }, {
+      archived: !(outreach.archived)
+    }).$promise;
+    promise.then(function(outreach) {
+      $scope.$evalAsync(function() {
+        var outreaches = $scope.student.outreaches
+        outreaches.map(function(oldOutreach, i) {
+            if(oldOutreach._id === outreach._id) {
+              outreaches[i].archived = outreach.archived;
+            }
+        });
+      });
+      toastr.info(
+        'The ' + outreach.type +
+        ' outreach has been' +
+        (outreach.archived === true ? ' archived.' : ' unarchived.'));
+    }, function(err) {
+      outreach.archived = oldValue;
+      toastr.error(err);
+    });
+  };
+  $scope.deleteOutreach = function(outreach) {
+    var model = outreach;
+    var deleteFn = function(model) {
+      Outreach.delete({id: model._id});
+      var outreaches = $scope.student.outreaches
+      outreaches.map(function(oldOutreach, i) {
+          if(oldOutreach._id === outreach._id) {
+            outreaches.splice(i, 1);
+          }
+      });
+      toastr.warning('Outreach ' + outreach.type +
+          ' has been deleted.');
+    };
+    Modal.confirmDelete(
+      'Delete Outreach',
+      'app/main/student/partial/modal.delete-outreach.html',
+      model,
+      deleteFn);
+  };
```

Comment:
To make this function async we just need to use the success and error callbacks:

```
  $scope.deleteOutreach = function(outreach) {
    var deleteFn = function(model) {
      Outreach.delete({id: model._id}, {}, function(deletedOutreach) {
        _.pull($scope.student.outreaches, {_id: deletedOutreach._id});
        toastr.warning('Outreach ' + outreach.type + ' has been deleted.');
      }, function(err) {
        console.log(err);
        toastr.error(err);
      });
    };
    Modal.confirmDelete(
      'Delete Outreach',
      'app/main/student/partial/modal.delete-outreach.html',
      outreach,
      deleteFn);
  };
```